tests/_support/tmpFiles.ts

```diff
@@ -1,11 +1,18 @@
 import { mkdirSync, mkdtempSync, existsSync } from 'fs';
 import { join } from 'path';
 
-export function tmpDirectory(): string {
+export function tmpDirectory(ext?: string): string {
 	if (!existsSync('.test')) {
 		mkdirSync('.test');
 	}
-	return mkdtempSync('.test/dir-');
+
+	const tmpDir = mkdtempSync('.test/dir-');
+
+	if (ext) {
+		return join(tmpDir, `index${ext}`);
```

Comment:
I understand why this was done because sometimes we want a filename. But it doesn't really follow that by passing an extension that it should make a temp directory and join it with a filename `index${ext}`. I would just move the join outside of this function to explicitly call out what's going on.